mapie/tests/test_regression.py

```diff
@@ -862,24 +884,106 @@ def test_fit_parameters_passing() -> None:
+    """
+    We want to verify that there are no interferences
+    with predict_params on the expected behavior of fit_params
+    Checks that underlying GradientBoosting
+    estimators have used 3 iterations only during boosting,
+    instead of default value for n_estimators (=100).
+    """
+    X_train, X_test, y_train, y_test = (
+        train_test_split(X, y, test_size=0.2, random_state=random_state)
+    )
+    custom_gbr = CustomGradientBoostingRegressor(random_state=random_state)
+    mapie_model = MapieRegressor(estimator=custom_gbr)
+    fit_params = {'monitor': early_stopping_monitor}
+    predict_params = {'check_predict_params': True}
+    mapie_model = mapie_model.fit(
+        X_train, y_train,
+        fit_params=fit_params, predict_params=predict_params
+    )
+
+    assert mapie_model.estimator_.single_estimator_.estimators_.shape[0] == 3
+    for estimator in mapie_model.estimator_.estimators_:
         assert estimator.estimators_.shape[0] == 3
 
 
+def test_predict_params_expected_behavior_unaffected_by_fit_params() -> None:
+    """
+    We want to verify that there are no interferences
+    with fit_params on the expected behavior of predict_params
+    Checks that the predictions on the training and test sets
+    are 0 for the model with predict_params and that this is not
+    the case for the model without predict_params
+    """
+    X_train, X_test, y_train, y_test = (
+        train_test_split(X, y, test_size=0.2, random_state=random_state)
+    )
+    custom_gbr = CustomGradientBoostingRegressor(random_state=random_state)
+    score = AbsoluteConformityScore(sym=True)
+    mapie_model = MapieRegressor(estimator=custom_gbr, conformity_score=score)
+    fit_params = {'monitor': early_stopping_monitor}
+    predict_params = {'check_predict_params': True}
+    mapie_model = mapie_model.fit(
+        X_train, y_train,
+        fit_params=fit_params,
+        predict_params=predict_params
+    )
+    y_pred = mapie_model.predict(X_test, **predict_params)
+
+    np.testing.assert_array_equal(mapie_model.conformity_scores_,
+                                  np.abs(y_train))
+    np.testing.assert_allclose(y_pred, 0)
+
+
+def test_invalid_predict_parameters() -> None:
+    """Test that invalid predict_parameters raise errors."""
+    custom_gbr = CustomGradientBoostingRegressor(random_state=random_state)
+    X_train, X_test, y_train, y_test = (
+        train_test_split(X, y, test_size=0.2, random_state=random_state)
+    )
+    mapie = MapieRegressor(estimator=custom_gbr)
+    predict_params = {'check_predict_params': True}
+    mapie_fitted = mapie.fit(X_train, y_train)
+
+    with pytest.raises(ValueError, match=(
+        fr".*Using 'predict_param' '{predict_params}' "
+        r"without using one 'predict_param' in the fit method\..*"
+        r"Please ensure one 'predict_param' "
+        r"is used in the fit method before calling predict\..*"
+    )):
+        mapie_fitted.predict(X_test, **predict_params)
+
+
```

Comment:
Why do we not check if a `predict_params` has not been given in `predict` despite being in the `fit`?

Reply:
Yes it's a good idea, I'm working on it